crates/router/src/routes/app.rs

```diff
@@ -1493,23 +1498,36 @@ impl BusinessProfile {
                             },
                         )),
                     )
-                    .service(web::resource("/deactivate_routing_algorithm").route(
-                        web::patch().to(|state, req, path| {
-                            routing::routing_unlink_config(
+                    .service(
+                        web::resource("/deactivate_routing_algorithm").route(web::patch().to(
+                            |state, req, path| {
+                                routing::routing_unlink_config(
+                                    state,
+                                    req,
+                                    path,
+                                    &TransactionType::Payment,
+                                )
+                            },
+                        )),
+                    )
+                    .service(web::resource("/routing_algorithm").route(web::get().to(
```

Comment:
Do not call the function directly here, define a route in routes and then call that here